src/main/java/de/unijena/cheminf/mortar/gui/controls/ToggleSwitch.java

```diff
@@ -0,0 +1,309 @@
+     * @param aRectangleOutline the color of the outline of the background.
+     * @param aCircleColor the color of the button.
+     * @param aCircleOutline the color of the outline of the button.
+     * @param aCircleShadowColor the color of the shadow around the button.
+     * @param aRectangleColorOn the color of the background when the switch is turned on.
+     */
+    public ToggleSwitch(boolean anInitialStateOfBooleanProperty, int aRectangleWidth, int aRectangleHeight, int aRectanglePositionX, int aCircleRadius,
+                        int aCirclePositionX, int aCirclePositionY, int aDropShadowRadius, double aDurationOfTransitionInSeconds, Color aRectangleColorOff,
+                        Color aRectangleOutline, Color aCircleColor, Color aCircleOutline, Color aCircleShadowColor, Color aRectangleColorOn){
+        this.switchStateBooleanProperty = new SimpleBooleanProperty(anInitialStateOfBooleanProperty);
+        this.switchBackground = new Rectangle(aRectangleWidth, aRectangleHeight);
+        this.switchBackground.setArcWidth(aRectangleHeight);
+        this.switchBackground.setArcHeight(aRectangleHeight);
+        this.switchBackground.setLayoutX(aRectanglePositionX);
+        this.switchBackground.setFill(aRectangleColorOff);
+        this.switchBackground.setStroke(aRectangleOutline);
+        this.switchButton = new Circle(aCircleRadius);
+        this.switchButton.setCenterX(aCirclePositionX);
+        this.switchButton.setCenterY(aCirclePositionY);
+        this.switchButton.setFill(aCircleColor);
+        this.switchButton.setStroke(aCircleOutline);
+        this.switchButton.setEffect(new DropShadow(aDropShadowRadius, aCircleShadowColor));
+        this.switchAnimation = new TranslateTransition(Duration.seconds(aDurationOfTransitionInSeconds));
+        this.switchAnimation.setNode(this.switchButton);
+        this.fillAnimation = new FillTransition(Duration.seconds(aDurationOfTransitionInSeconds));
+        this.switchTransition = new ParallelTransition(this.switchAnimation, this.fillAnimation);
+        this.switchAnimation.setNode(this.switchButton);
+        this.fillAnimation.setShape(this.switchBackground);
+        this.getChildren().addAll(this.switchBackground, this.switchButton);
+        //Listener
+        this.switchStateBooleanProperty.addListener((observable, oldValue, newValue) -> {
+            boolean tmpIsOn = newValue.booleanValue();
+            this.switchAnimation.setToX(tmpIsOn ? (aRectangleWidth - (2 * aCircleRadius)) : 0); //?
+            this.fillAnimation.setFromValue(tmpIsOn ? aRectangleColorOff : aRectangleColorOn);
+            this.fillAnimation.setToValue(tmpIsOn ? aRectangleColorOn : aRectangleColorOff);
+            this.switchTransition.play();
+        });
+        //Mouse listener.
+        this.setOnMouseClicked(event -> this.switchStateBooleanProperty.set(!this.switchStateBooleanProperty.get()));
```

Comment:
The constructor above that uses all the default values should simply call this constructor here using the default values as parameters to avoid code duplications.

Reply:
Maybe add one constructor more only with those parameters that people are more likely wanting to adjust, e.g. the colors. Use defaults for the rest.